content/root/best-practice-guide.textile

```diff
@@ -5,45 +5,83 @@ index: 30
+
+If you prefer to use a single auth strategy across all third-party services, you can use "JWTs":/core-features/authentication#ably-jwt-process to authenticate with Ably.
+
+h3(#using-api-key-client-side). Using the API key on the client side
+
+"Basic authentication":/core-features/authentication#basic-authentication is the simplest method to authenticate your clients on the Ably platform, but this exposes your API key in the source code. Therefore, Ably recommends that you create an auth server and use "token authentication":/core-features/authentication#token-authentication to authenticate your clients. Using this approach, only your auth server has access to your API key, protecting it from being misused. "Read more":/core-features/authentication#selecting-auth about selecting an authentication mechanism.
 
-With Ably's "Presence":/realtime/presence feature, your clients can attach to a presence channel and announce to the rest of the users that they are present on a channel. This is very useful, for example, in chat apps to indicate a member is online, or with IoT devices to indicate a device is online and connected. Clients can subscribe to realtime presence changes on channels, such as members entering or leaving, and respond to these events accordingly.
+h3(#ensuring-tokens-are-auto-renewable). Renewing tokens
+
+Tokens are short-lived, that is, they expire after a given time. You must ensure that the client can request a new token when it needs one. The Ably Client Library SDKs provide two different mechanisms to obtain tokens automatically: "@authUrl@":/realtime/authentication#token-upgrading or "@authCallback@":/realtime/authentication#token-upgrading. You specify which one to use in the "ClientOptions":/realtime/authentication#token-authentication object.
 
 h3(#rest-or-realtime). Realtime vs REST
 
-Ably provides two types of client libraries. "Realtime":/realtime libraries maintain a persistent connection to Ably and can be considered stateful, whereas "REST":/rest libraries issue HTTP requests to the Ably platform and are stateless.
+Ably Client Library SDKs may implement two interfaces: realtime and REST. The "Realtime":/realtime interface maintains a persistent connection to Ably and can be considered stateful. The "REST":/rest interface issues HTTP requests to the Ably platform and is stateless. The REST interface is therefore lighter weight because it does not need to maintain state. 
+
+Depending on your use case, you may only need to use the REST interface on your server. For example, the server may only need to authenticate with Ably, and then act as an auth server for clients. You might then use the realtime interface on your client, so it can both subscribe and publish messages.
```

Comment:
```suggestion
Depending on your use case, you may only need to use the REST interface on your server. For example, the server may only need to authenticate with Ably, and then act as an auth server for clients. You might then use the realtime interface on your client, so it can both subscribe to and publish messages.
```